tools/clang-func-mapping/ClangFnMapGen.cpp

```diff
@@ -113,34 +80,30 @@ void MapFunctionNamesConsumer::handleDecl(const Decl *D) {
     return;
 
   if (const auto *FD = dyn_cast<FunctionDecl>(D)) {
-    if (const Stmt *Body = FD->getBody()) {
-      std::string MangledName = getMangledName(FD);
-      const SourceManager &SM = Ctx.getSourceManager();
-      if (CurrentFileName.empty()) {
-        StringRef SMgrName =
-            SM.getFileEntryForID(SM.getMainFileID())->getName();
-        char *Path = realpath(SMgrName.str().c_str(), nullptr);
-        CurrentFileName = Path;
-        free(Path);
+    if (FD->isThisDeclarationADefinition()) {
+      if (const Stmt *Body = FD->getBody()) {
+        std::string MangledName = getMangledName(FD);
+        const SourceManager &SM = Ctx.getSourceManager();
+        if (CurrentFileName.empty()) {
+          StringRef SMgrName =
+              SM.getFileEntryForID(SM.getMainFileID())->getName();
+          char *Path = realpath(SMgrName.str().c_str(), nullptr);
+          CurrentFileName = Path;
+          free(Path);
+        }
+
+        switch (FD->getLinkageInternal()) {
+        case ExternalLinkage:
+        case VisibleNoLinkage:
+        case UniqueExternalLinkage:
+          if (SM.isInMainFile(Body->getLocStart())) {
+            // DefinedFuncsStr << "!";
```

Comment:
I think we can remove this comment. Unnecessary load for the future generations.